packages/cli/src/services/user.service.ts

```diff
@@ -249,6 +251,18 @@ export class UserService {
 							const newUser = transactionManager.create(User, { email, role });
 							const savedUser = await transactionManager.save<User>(newUser);
 							createdUsers.set(email, savedUser.id);
+							const savedProject = await transactionManager.save<Project>(
+								transactionManager.create(Project, {
+									type: 'personal',
+								}),
+							);
+							await transactionManager.save<ProjectRelation>(
+								transactionManager.create(ProjectRelation, {
+									projectId: savedProject.id,
+									userId: savedUser.id,
+									role: 'project:personalOwner',
+								}),
+							);
```

Comment:
question: Would that also work for the first user that is created for an instance? 

Not sure which endpoint that is using:

![image](https://github.com/n8n-io/n8n/assets/927609/699d8300-f249-463e-9947-900639283eb9)

I think that's here:
https://github.com/n8n-io/n8n/blob/647bac07a86d3819ff8933a3a9d077e075571c1e/packages/cli/src/controllers/owner.controller.ts#L36-L97

I'm wondering if both the `OwnerController.setupOwner` could `UserService.inviteUsers` could use something like `UserService.save` and the logic for creating the auxiliary resources lives in there.

What do you think?

Reply:
If we do this, we can also backport the refactor to master and thus make the diff for `feature/rbac` smaller. Just a thought.

Reply:
This will work for the initial owner account because that account is created as part of an earlier migration. Our migration will pick up any existing users (in this case the instance owner if it's a fresh instance) and will create a personal project for them.